.github/workflows/format.yml

```diff
@@ -0,0 +1,26 @@
+name: Clang format
+
+on: [pull_request]
+
+jobs:
+  clang-format-8:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Run clang-foramt on changed files
+        run: |
+          set -x
+          git fetch origin ${{ github.event.pull_request.base.ref }}
+          git fetch origin pull/${{ github.event.pull_request.number }}/head:${{ github.event.pull_request.head.ref }}
+          BASE_COMMIT=$(git rev-parse ${{ github.event.pull_request.base.sha }})
+          COMMIT_FILES=$(git diff --name-only ${BASE_COMMIT} | grep -i -v LinkDef)
+          RESULT_OUTPUT=$(git-clang-format-8 --commit ${BASE_COMMIT} --diff --binary $(which clang-format-8) ${COMMIT_FILES})
+          if [ "$RESULT_OUTPUT" == "no modified files to format" ] || [ "$RESULT_OUTPUT" == "clang-format did not modify any files" ]; then
```

Comment:
What if the text changes with a future version? Maybe it would be better to check the exit code? (if it is actually non-zero when there are no files to format)

Reply:
Unfortunately the tool returns the same value in both cases. :(